### PR TITLE
Deprecate `maxLengthEnforced` for text fields

### PR DESCRIPTION
**Pull request by AlexV525**
## Description

Further work for #68086 . Deprecate `maxLengthEnforced` for text fields, and some nit picking update.

## Tests

I added the following tests:

* _LengthLimitingTextInputFormatter get enforcement from target platform_ for text formatter test.

## Checklist

Before you create this PR, confirm that it meets all requirements listed below by checking the relevant checkboxes (`[x]`). This will ensure a smooth and quick review process.

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I signed the [CLA].
- [x] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] All existing and new tests are passing.
- [x] The analyzer (`flutter analyze --flutter-repo`) does not report any problems on my PR.
- [x] I am willing to follow-up on review comments in a timely manner.

## Breaking Change

Did any tests fail when you ran them? Please read [Handling breaking changes].

- [x] No, no existing tests failed, so this is *not* a breaking change.
- [ ] Yes, this is a breaking change.

[issue database]: https://github.com/flutter/flutter/issues
[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Test Coverage]: https://github.com/flutter/flutter/wiki/Test-coverage-for-package%3Aflutter
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[Features we expect every widget to implement]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#features-we-expect-every-widget-to-implement
[CLA]: https://cla.developers.google.com/
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[Handling breaking changes]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes

### CODE REVIEW

**Comment by AlexV525** [comment]
Hmm, the commit seems not rolled in yet. Any ideas?

File: packages/flutter/lib/src/cupertino/text_field.dart [hunk 1]
```diff
@@ -805,7 +820,7 @@ class _CupertinoTextFieldState extends State<CupertinoTextField> with Restoratio
   FocusNode get _effectiveFocusNode => widget.focusNode ?? (_focusNode ??= FocusNode());
 
   MaxLengthEnforcement get _effectiveMaxLengthEnforcement => widget.maxLengthEnforcement
-    ?? LengthLimitingTextInputFormatter.inferredDefaultMaxLengthEnforcement;
+    ?? LengthLimitingTextInputFormatter.getDefaultMaxLengthEnforcement();
```
Comment (LongCatIsLooong) [inline packages/flutter/lib/src/cupertino/text_field.dart:823]: like the new name :+1:

**Review by LongCatIsLooong — COMMENTED** [review]
The PR looks good to me. Thank you for your patience and the effort put into the series of pull requests!

File: packages/flutter/lib/src/cupertino/text_field.dart (hunk 1 above)
Comment (justinmc) [reply packages/flutter/lib/src/cupertino/text_field.dart:823]: Is this related to the deprecation at all or just an improvement?

**Review by justinmc — APPROVED** [review]
LGTM, just one nit about an optional parameter.

I see that you updated the gallery (https://github.com/flutter/gallery/pull/358) but it's still failing here.  I'm not sure what needs to happen for the tests to get updated here...

File: packages/flutter/lib/src/cupertino/text_field.dart (hunk 1 above)
Comment (AlexV525) [reply packages/flutter/lib/src/cupertino/text_field.dart:823]: It's an improvement, that the style guide say we preferred `getXXX` for method.

**Comment by AlexV525** [comment]
After the https://github.com/flutter/tests/pull/72 get's merged, customer tests will be succeed.

File: packages/flutter/lib/src/services/text_formatter.dart [hunk 2]
```diff
@@ -432,11 +433,13 @@ class LengthLimitingTextInputFormatter extends TextInputFormatter {
   ///    [MaxLengthEnforcement.truncateAfterCompositionEnds]. These platforms
   ///    allow the composition to exceed by default.
   /// {@endtemplate}
-  static MaxLengthEnforcement get inferredDefaultMaxLengthEnforcement {
+  static MaxLengthEnforcement getDefaultMaxLengthEnforcement({
+    TargetPlatform? platform,
```
[the remark this answers is not in the document]
Comment (AlexV525) [reply packages/flutter/lib/src/services/text_formatter.dart:437]: If I remember it correctly, `defaultTargetPlatform` is a getter, so it cannot be the default value for arguments.

[the remark this answers is not in the document]
Comment (AlexV525) [reply packages/flutter/lib/src/services/text_formatter.dart:437]: A position optional param or a named optional param both SGTM, I'll take your suggestion.

**Pull request merged**